### PR TITLE
Titan/ORNL: Update Modules

### PR DESCRIPTION
**Pull request by ax3l**
This pull requests updates the `picongpu.profile` example that we provide for Titan/ORNL.

I also added an example profile that can be used for python on compute nodes, similar to my gist at https://gist.github.com/ax3l/7d1fa52bb7f39a657920 .

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
Can we add a link in `pythonOnComp.profile.example` to your Gist (https://gist.github.com/ax3l/7d1fa52bb7f39a657920) or the Gist data as `README.md`

**Comment by ax3l** [comment]
good idea, I just added the link in the commit message so far

**Comment by ax3l** [comment]
I have to update this later on, I found out yesterday on Rhea, that using [Python virtual environments](http://docs.python-guide.org/en/latest/dev/virtualenvs/) is even more gentle then setting all the paths in a profile (and allows easier switching between parallel installs)

File: src/picongpu/submit/titan-ornl/pythonOnRhea.profile.example [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#
+# now run for setting up the python virtual environment
+#   mkdir -p $PROJWORK/$proj/python-venv
+#   cd $PROJWORK/$proj
+#   virtualenv rhea
+#   source $PROJWORK/$proj/python-venv/rhea/bin/activate
+#   which pip
+#   pip install adios
+#   pip install six
+#   pip install matplotlib
```
Comment (ax3l) [inline src/picongpu/submit/titan-ornl/pythonOnRhea.profile.example]: add for pyDive support:
```batch
module load ipython/3.0.0

# and once for initial setup 
pip install pyzmq
pip install mpi4py
pip install h5py
pip install psutil
pip install pyDive
```

Comment (psychocoderHPC) [inline src/picongpu/submit/titan-ornl/pythonOnRhea.profile.example]: please add this to a persistent file

Comment (ax3l) [inline src/picongpu/submit/titan-ornl/pythonOnRhea.profile.example]: yep, just written as a reminder that this is still *wip* :)

File: src/picongpu/submit/titan-ornl/picongpu.profile.example [hunk 2]
```diff
@@ -9,6 +10,8 @@ module swap PrgEnv-pgi PrgEnv-gnu
 #   (CMake likes to unwrap the Cray wrappers)
 export CC="`which cc` -target=compute_node"
```
Comment (ax3l) [inline src/picongpu/submit/titan-ornl/picongpu.profile.example]: Note from ORNL support:
using
```bash
cmake -DCMAKE_CXX_COMPILER=`which CC` [...]
```
might result in the same behavior (CMake should not unwrap the compiler-wrappers to their underlying compiler such as gcc).

**Pull request merged**